### PR TITLE
Add documentation for other Button types

### PR DESCRIPTION
**Pull request by rurikoaraki**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Adding documentation for FAB, Toggle, and Compound buttons.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Button/src/CompoundButton/SPEC.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The space for the icon is larger in a `CompoundButton`. The `CompoundButton` is designed so that it can hold an illustration in the place of an icon.
+
+## API
+
+### Slots
+
+- `root` - The outer container representing the `Button` itself that wraps everything passed via the `children` prop.
+- `icon` - If specified, renders an `icon` either before or after the `children` as specified by the `iconPosition` prop.
+- `loader` - If specified, renders a `loader` before the `icon` and `children` while the `loading` flag is set to `true`.
```
Comment (ValentinaKozlova) [inline packages/experimental/Button/src/CompoundButton/SPEC.md:36]: I remember comment from Sunmin that there won't be icons when we have a loading button: "Loading button will not have any icon next to it. It'll always have spinner + text 'loading'"

**Review by ValentinaKozlova — APPROVED** [review]

File: packages/experimental/Button/src/CompoundButton/SPEC.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Compound Button
+
+## Background
+
+The `CompoundButton` is a `Button` that is designed to show an extra line of text.
+
+## Sample Code
+
+Basic examples:
```
Comment (samuelfreiberg) [inline packages/experimental/Button/src/CompoundButton/SPEC.md:9]: Is it possible to put an image here of what one of these examples looks like? I know it's pretty self-explanatory, but I always think seeing it visually along with the code snippet helps :)

File: packages/experimental/Button/src/FAB/SPEC.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# FAB
+
+## Background
+
+The `FAB` is a `Button` that is designed for mobile scenarios. This button only renders on mobile platforms.
```
Comment (samuelfreiberg) [inline packages/experimental/Button/src/FAB/SPEC.md:5]: So this includes iOS and Android?

File: packages/experimental/Button/src/CompoundButton/SPEC.md (hunk 1 above)
Comment (rurikoaraki) [reply packages/experimental/Button/src/CompoundButton/SPEC.md:36]: Thanks, I'll update the text

File: packages/experimental/Button/src/CompoundButton/SPEC.md (hunk 2 above)
Comment (rurikoaraki) [reply packages/experimental/Button/src/CompoundButton/SPEC.md:9]: Sure, I'll add some visuals.

My hope is that in the long run the website which will house the documentation will allow us to have code snippets with visual examples of the button. Documentation in the repo is a short-term thing.

File: packages/experimental/Button/src/FAB/SPEC.md (hunk 3 above)
Comment (rurikoaraki) [reply packages/experimental/Button/src/FAB/SPEC.md:5]: Yeah, FAB is just iOS and Android.

**Review by samuelfreiberg — APPROVED** [review]

**Pull request merged**